### PR TITLE
Add video views API

### PR DESCRIPTION
**Pull request by jmaupetit**
## Purpose

The first (easiest) indicator we choose to implement is a video views endpoint.

## Proposal

To be able to version our API, the root app mounts apps from different versions. Starting with v1, we now connect to an Elasticsearch cluster and add support for video views calculation.

An example request would look like:

```
curl "http://localhost:8100/api/v1/video/{video_id}/views"
```

with `{video_id}` corresponding to the URL encoded video IRI, _e.g._:

```
curl "http://localhost:8100/api/v1/video/uuid%3A%2F%2F8d386f48-3baa-4acf-8a46-0f2be4ae243e/views"
```

The (truncated) response would look like:

```json
{
  "total": 73,
  "daily_views": [
    {
      "day": "2023-02-08",
      "views": 1
    },
    {
      "day": "2023-02-09",
      "views": 3
    },
    {
      "day": "2023-03-07",
      "views": 4
    }
  ]
}
```

### CODE REVIEW

**Comment by quitterie-lcs** [comment]
Backend can not be run correctly. Some environment variables are missing. 
The following variable should be documented in the project when tested :

```
# Elasticsearch
WARREN_ES_HOSTS=http://elasticsearch:9200
WARREN_ES_INDEX=statements
```

File: src/backend/tests/api/v1/test_video.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+import pytest
+from httpx import AsyncClient
+
+from warren.api import app
+from warren.conf import settings
+
+
+@pytest.mark.asyncio
+async def test_views_video_id_path(es):
```
Comment (quitterie-lcs) [inline src/backend/tests/api/v1/test_video.py:11]: Why not parametrize this test ?

File: src/backend/warren/api/v1/video/__init__.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
+"""Warren API v1 video router."""
```
Comment (wilbrdt) [inline src/backend/warren/api/v1/video/__init__.py:1]: Just a question: will the API routes stay in each __init__.py files?

**Review by wilbrdt — COMMENTED** [review]
Alt text

File: src/backend/warren/api/v1/video/__init__.py (hunk 2 above)
Comment (jmaupetit) [reply src/backend/warren/api/v1/video/__init__.py:1]: Nope, we'll categorize/move them in separated sub-modules I think.

File: src/backend/tests/api/v1/test_video.py (hunk 1 above)
Comment (jmaupetit) [reply src/backend/tests/api/v1/test_video.py:11]: You are right. Will do this.

File: src/backend/warren/api/v1/video/__init__.py [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+                "must": [
+                    {"match": {"verb.display.en-US.keyword": "played"}},
+                    {"match": {"object.id.keyword": video_id}},
+                ],
+                "filter": [
+                    {
+                        "range": {
+                            (
+                                "result.extensions."
+                                "https://w3id.org/xapi/video/extensions/time"
```
[the remark this answers is not in the document]
Comment (jmaupetit) [reply src/backend/warren/api/v1/video/__init__.py:44]: Yup.

File: src/backend/warren/api/v1/video/__init__.py [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+                "date_histogram": {
+                    "field": settings.ES_INDEX_TIMESTAMP_FIELD,
+                    "calendar_interval": "day",
+                }
+            }
+        },
+        "size": 0,
+    }
+    docs = await backends.es_client.search(
+        **query_params,
```
[the remark this answers is not in the document]
Comment (jmaupetit) [reply src/backend/warren/api/v1/video/__init__.py:71]: :thinking: there is no `body` argument in this request.

File: src/backend/tests/fixtures/backends.py [hunk 5]
```diff
@@ -0,0 +1,45 @@
+"""Warren test fixtures for supported backends."""
+
+import pytest
+from elasticsearch import AsyncElasticsearch, BadRequestError
+
+from warren import backends
+from warren.conf import settings
+
+pytestmark = pytest.mark.anyio
```
[the remark this answers is not in the document]
Comment (jmaupetit) [reply src/backend/tests/fixtures/backends.py]: Good catch!

File: src/backend/warren/api/v1/video/__init__.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (wilbrdt) [reply src/backend/warren/api/v1/video/__init__.py:71]: Whoops you are right, I'm hallucinating things!

**Review by wilbrdt — APPROVED** [review]
Alt text

**Pull request merged**